### DISCUSSION TITLE
[Build]

### DISCUSSION THREAD

**Discussion by EricKong1985** [PyMySQL/mysqlclient#658] [Build issue]
### What OS and which version do you use?

[root@yisu-64de4a1ae2a3e ~]# cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7

### How did you installed mysql client library?

 pip3 install mysqlclient

### Output from `pkg-config --cflags --libs mysqlclient`

Package mysqlclient was not found in the pkg-config search path.
Perhaps you should add the directory containing `mysqlclient.pc'
to the PKG_CONFIG_PATH environment variable
No package 'mysqlclient' found

[root@yisu-64de4a1ae2a3e ~]# sudo yum install mysql-devel
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Package 1:mariadb-devel-5.5.68-1.el7.x86_64 already installed and latest version
Nothing to do

### How did you tried to install mysqlclient?

pip install mysqlclien

### Output of building mysqlclient

```bash
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/include/openssl11 -I/usr/include/openssl11 -I/usr/include/openssl11 -fPIC "-Dversion_info=(2, 2, 0, 'final', 0)" -D__version__=2.2.0 -I/usr/python39/include/python3.9 -c src/MySQLdb/_mysql.c -o build/temp.linux-x86_64-cpython-39/src/MySQLdb/_mysql.o -I/usr/include/mysql -std=c99
      src/MySQLdb/_mysql.c: In function ‘_mysql_ConnectionObject_Initialize’:
      src/MySQLdb/_mysql.c:536:13: warning: implicit declaration of function ‘mysql_optionsv’ [-Wimplicit-function-declaration]
                   mysql_optionsv(&(self->connection), MYSQL_OPT_SSL_ENFORCE, (void *)&enforce_tls);
                   ^
      src/MySQLdb/_mysql.c:536:49: error: ‘MYSQL_OPT_SSL_ENFORCE’ undeclared (first use in this function)
                   mysql_optionsv(&(self->connection), MYSQL_OPT_SSL_ENFORCE, (void *)&enforce_tls);
                                                       ^
      src/MySQLdb/_mysql.c:536:49: note: each undeclared identifier is reported only once for each function it appears in
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for mysqlclient
Failed to build mysqlclient
ERROR: Could not build wheels for mysqlclient, which is required to install pyproject.toml-based projects
```

**Comment by methane** [comment]
Your environment is tooooo old.
Upgrade your OS, install newer libraries, or use old mysqlclient.
